src/d_player.h

```diff
@@ -636,6 +636,9 @@ class player_t
 	// This player is chatting.
 	bool		bChatting;
 
+	// [AK] The last few chat messages we received from this player.
+	TArray<FString>		ChatMessages;
```

Comment:
I'm not sure whether the chat messages should be associated with the player directly. It's quite counterintuitive to me that one has to copy chat messages if a player is reborn. I assume you put them here since you didn't want do associate them with the client so that this also works offline?

Reply:
I want to make sure this could work offline and online (both on the client's and server's ends). Initially, I thought about adding an `FString` array to `chat.cpp` that holds the chat messages for all players + one extra element for the server. If a player left, that element would need to be emptied. However, I was hesitant about including `chat.h` to `p_acs.cpp` so I ended up adding the array directly into `player_t`.

Reply:
Putting this into a `RingBuffer` in `chat.cpp` in your latest patch looks good.